src/components/help/help.component.tsx

```diff
@@ -9,54 +10,102 @@ import Events from "../../__internal__/utils/helpers/events";
 import { TooltipContext } from "../../__internal__/tooltip-provider";
 import { filterStyledSystemMarginProps } from "../../style/utils";
 import { HELP_POSITIONS } from "./help.config";
+import { TooltipPositions } from "../tooltip/tooltip.config";
 import guid from "../../__internal__/utils/helpers/guid";
 
-const marginPropTypes = filterStyledSystemMarginProps(
-  styledSystemPropTypes.space
-);
+export interface HelpProps extends MarginProps {
+  /** Overrides the default 'as' attribute of the Help component */
+  as?: keyof JSX.IntrinsicElements;
+  /** Aria label */
+  ariaLabel?: string;
+  /** The message to be displayed within the tooltip */
+  children?: React.ReactNode;
+  /** [Legacy] A custom class name for the component. */
+  className?: string;
+  /** The unique id of the component (used with aria-describedby for accessibility) */
+  helpId?: string;
+  /** A path for the anchor */
+  href?: string;
+  /** A boolean received from IconWrapper */
+  isFocused?: boolean;
+  /** Overrides the default tabindex of the component */
+  tabIndex?: number;
+  /** Override background color of the Tooltip, provide any color from palette or any valid css color value. */
+  tooltipBgColor?: string;
+  /** Override font color of the Tooltip, provide any color from palette or any valid css color value. */
+  tooltipFontColor?: string;
+  /** Overrides the default flip behaviour of the Tooltip,
+   *  must be an array containing some or all of ["top", "bottom", "left", "right"]
+   * (see https://popper.js.org/docs/v2/modifiers/flip/#fallbackplacements)
+   */
+  tooltipFlipOverrides?: TooltipPositions[];
+  /** Id passed to the tooltip container, used for accessibility purposes */
+  tooltipId?: string;
+  /** Position of tooltip relative to target */
+  tooltipPosition?: TooltipPositions;
+  /** Help Icon type */
+  type?: IconType;
+  // any has been used here to allow rest props to be spread
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  [key: string]: any;
+}
 
-const Help = ({
+export const Help = ({
+  as,
+  ariaLabel = "help",
+  children,
   className,
   href,
   helpId,
-  children,
-  tabIndex,
-  as,
-  tooltipPosition,
   isFocused,
-  type,
-  tooltipId,
+  tabIndex = 0,
   tooltipBgColor,
   tooltipFontColor,
   tooltipFlipOverrides,
-  ariaLabel = "help",
+  tooltipId,
+  tooltipPosition = "top",
+  type = "help",
   ...rest
-}) => {
+}: HelpProps): JSX.Element => {
   const defaultTooltipId = useRef(guid());
-  const helpElement = useRef(null);
+  const helpElement = useRef<HTMLDivElement>(null);
   const [isTooltipVisible, updateTooltipVisible] = useState(false);
   const { helpAriaLabel } = useContext(TooltipContext);
+  const flipBehaviourCheck = useMemo(
```

Comment:
Would the typescript compiler be able to handle this check for us? Or is it possible for the values in `tooltipFlipOverrides` to change at runtime?

Reply:
If the consuming project is TypeScript based then we wouldn't need the check but if they are using JS we will need to have a way of constraining the values passed in to `tooltipFlipOverrides`

Reply:
I think that we could remove this check, as this will be already checked in the Icon component to which this prop is passed

Reply:
Valid point, I'll block this ticket until your refactor is merged 👍